api/src/main/java/org/openmrs/api/impl/ObsServiceImpl.java

```diff
@@ -180,28 +179,15 @@ private Obs saveObsNotDirty(Obs obs, String changeMessage) {
 		}
 
 		ObsService os = Context.getObsService();
-		List<Obs> toRemove = new ArrayList<>();
-		List<Obs> toAdd = new ArrayList<>();
 		for (Obs o : obs.getGroupMembers(true)) {
 			if (o.getId() == null) {
 				os.saveObs(o, null);
 			} else {
-				Obs replacement = os.saveObs(o, changeMessage);
-				if(!replacement.equals(o)){
```

Comment:
I.e you need a flag that you set to true if we have a replacement and then and check it below to determine is a refresh is needed